pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -206,3 +206,8 @@ def test_multiindex_with_na_missing_key(self):
         )
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
+
+    def test_multiindex_from_tuples_with_nan(self):
+        # GH#23578
+        multi_idx = MultiIndex.from_tuples([("a", "b", "c"), np.nan, ("d", "", "")])
+        assert isinstance(multi_idx, MultiIndex)
```

Comment:
Can you construct 

`expected = MultiIndex.from_tuples([("a", "b", "c"), (np.nan, np.nan, np.nan), ("d", "", "")])`

and then call

`tm.assert_index_equal_result, expected`

Reply:
Done